jwt-react/src/components/AuthenticatedPingTest/AuthenticatedPingTest.js

```diff
@@ -0,0 +1,32 @@
+import React, { useContext, useState } from 'react';
+import { UserContext } from '../../contexts/userContext.js'
+import { ping } from '../../api/rest.js'
+
+function AuthenticatedPingTest() {
+  const [pingResult, setPingResult] = useState();
+  const {user, setUser, isUserLoggedIn} = useContext(UserContext)
+
+  const testPing = () => {
+    ping()
+    .then((data)=>{
+      setPingResult(data.id)
+    }).catch((error)=> {
+      setUser(null);
+    });
+  }
+
+  return (<div>
+      {isUserLoggedIn() ?
+        <React.Fragment>
+          <h2>Authenticated Request Test</h2>
+          <button onClick={testPing}>Test Ping</button>
+          <p>Ping Result: {pingResult} </p>
+        </React.Fragment>
+        :
+        <p>Please Login to see the ping test</p>
```

Comment:
Only read on if the following is **IN**correct: if the user cannot hit the ping url, the Django server returns "Please login to see the ping test". Imo, I don't think that's true:

---

Hm "Please login to see the ping test"... if the user is logged out and the user is redirected to the log in page in which the `?next=` query parameter shows the page the user should go to on login, I don't think most people would like to see "Please login to see the ping test" (note "Login" should be lowercase).

In other words, this doesn't really make too much sense to me. If the user is forced to **RE**login, they shouldn't see "ping test" but rather the regular log in page.